*: replaced ignition and cloud config with container linux configs

dgonyeo opened the pull request:
Here's an incomplete pass at changing all the OS docs over to using container linux configs instead of ignition or cloud configs.

Everything alphabetically before `iscsi.md` should be good to go, with the exception of `cluster-architecture.md`, which needs a complete rewrite.

dgonyeo commented:
This should now be complete, and ready for review

dgonyeo commented on os/customizing-sshd.md:
```diff
@@ -1,57 +1,56 @@
-      AllowUsers core
-      PasswordAuthentication no
-      ChallengeResponseAuthentication no
+          PermitRootLogin no
```
You mean to inlcude the default config and append lines to it? `Include` in what context? ct/ignition don't have that functionality, and I don't see an `Include` directive in https://linux.die.net/man/5/sshd_config.

dgonyeo commented:
Found and replaced the ignition and cloud config references outside of the `os` dir, which I had forgotten about initially. Need to rewrite large parts of `etcd/tls-etcd-clients.md`. Going to do the standard CLC thing you mentioned, still getting to it.

dgonyeo commented:
Just did what I probably should've done in the beginning: pulled in [ct's docs test](https://github.com/coreos/container-linux-config-transpiler/blob/master/doc/doc_test.go) and tweaked it for here. All markdown files in this repo are now parsed for CLC blocks, and the found blocks are run through the parser and converter. Any warnings or errors cause the test to fail.

There's no CI for it on this repo, but at the very least it's something that docs authors can run when they modify/add CLC blocks.

crawford commented on os/mounting-storage.md:
```diff
@@ -1,17 +1,16 @@
     - name: media-ephemeral.mount
-      command: start
-      content: |
+      enable: true
```
There is no install section.

crawford commented on os/notes-for-distributors.md:
```diff
@@ -18,9 +18,30 @@ gpg --verify coreos_production_openstack_image.img.bz2.sig
-Use cloud-config to handle platform specific configuration such as custom networking, running an agent on the machine or injecting files onto disk. Container Linux will automatically parse and execute `/usr/share/oem/cloud-config.yml` if it exists. Your cloud-config should create additional units that process user-provided metadata, as described below.
+### Ignition
+
+[Ignition][ignition] is a tool that acquires a JSON config file when a machine first boots, and uses this config to perform tasks such as formatting disks, creating files, modifying and creating users, and adding systemd units. How Ignition acquires this config file varies per-platform, and it is highly recommended that providers ensure Ignition has [support for their platform][ign-platforms].
```
The method by which Ignition acquires this config varies per platform.

crawford commented on os/notes-for-distributors.md:
```diff
@@ -18,9 +18,30 @@ gpg --verify coreos_production_openstack_image.img.bz2.sig
+
+[Ignition][ignition] is a tool that acquires a JSON config file when a machine first boots, and uses this config to perform tasks such as formatting disks, creating files, modifying and creating users, and adding systemd units. How Ignition acquires this config file varies per-platform, and it is highly recommended that providers ensure Ignition has [support for their platform][ign-platforms].
+
+Ignition has a default config for each supported platform that is appended to any user provided config. Use Ignition to handle platform specific configuration such as custom networking, running an agent on the machine, or injecting files onto disk.
```
This first statement is an implementation detail and will be removed eventually.

crawford commented on os/other-settings.md:
```diff
@@ -8,93 +8,42 @@ Most Linux kernel modules get automatically loaded as-needed but there are a som
-The following section demonstrate how to provide module options when
-loading. After these configs are processed, the dummy module is loaded
-into the kernel, and five dummy interfaces are added to the network stack.
+The following section demonstrate how to provide module options when loading. After these configs are processed, the dummy module is loaded into the kernel, and five dummy interfaces are added to the network stack.
```
demonstrates

dgonyeo commented on os/booting-on-openstack.md:
````diff
@@ -72,41 +72,37 @@ $ glance image-create --name Container-Linux \
-ssh_authorized_keys:
-  # include one or more SSH public keys
-  - ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDGdByTgSVHq.......
+```container-linux-config:openstack-metadata
````
The line immediately after this block reads:
> The `{PRIVATE_IPV4}` and `{PUBLIC_IPV4}` substitution variables are fully supported in cloud-config on OpenStack deployments using the metadata service. Unfortunately systems relying on config drive are currently unsupported.

Is that sufficient?

crawford replied:
This isn't cloud-config though.

dgonyeo commented:
Just pushed some stuff:
- fixed errors in the ec2 CLC
- updated the digitalocean screenshot to be of an ignition config instead of a cloud config
- finished the `etcd/tls-etcd-clients.md` file
- switched all ignition-supported cloud providers to using the same, heavily commented CLC that starts etcd. This relies on https://github.com/coreos/container-linux-config-transpiler/pull/54, which is not yet in a release. I just tested booting a machine in ec2 with this CLC, and it behaved as expected.

crawford commented on os/customizing-sshd.md:
```diff
@@ -1,57 +1,56 @@
-      FreeBind=true
-      Accept=yes
+    - name: sshd.socket
+      enable: true
```
No install section.

crawford commented on os/mounting-storage.md:
```diff
@@ -66,13 +64,12 @@ Container Linux [561.0.0](https://coreos.com/releases/#561.0.0) and later are in
     - name: format-var-lib-docker.service
-      command: start
-      content: |
+      enable: true
```
No install section.

crawford commented on os/power-management.md:
```diff
@@ -21,17 +21,14 @@ modprobe cpufreq_conservative
-      command: start
-      runtime: true
-      content: |
+      enable: true
```
No install section.

crawford commented on os/scheduling-tasks-with-systemd-timers.md:
```diff
@@ -39,56 +39,31 @@ You can also create timer with different name, i.e. `task.timer`. In this case y
     - name: date.timer
-      command: start
-      content: |
+      enable: true
```
No install section.

crawford commented on etcd/etcd-live-cluster-reconfiguration.md:
````diff
@@ -4,29 +4,30 @@ This document describes the reconfiguration or recovery of an etcd cluster runni
-    discovery: https://discovery.etcd.io/<token>
+```container-linux-config
+etcd:
+  version: 3.1.4
````
We were going to toss the version, right?

crawford commented on os/quickstart.md:
````diff
@@ -40,29 +40,22 @@ CoreOS (beta)
-    discovery: https://discovery.etcd.io/<token>
+```container-linux-config
+etcd:
+  version: 3.1.4
````
Version.
